Sources/swiftML/KMeans.swift

```diff
@@ -0,0 +1,240 @@
+import TensorFlow
+
+/// K-Means Clustering
+public class KMeans {
+    
+    public var nClusters: Int
+    public var maxIterations: Int
+    // Centroids of Clsuters.
+    public var centroids : Tensor<Float>
+    // Inertia is the sum of square distances of samples to their closest cluster center.
+    public var inertia: Tensor<Float>
+    // Predicted cluster labels.
+    public var labels: Tensor<Int32>
+    public var initializer: String
+    public var seed: Int64
+  
+    /// Creates a logistic regression model.
+    ///
+    /// - Parameters
+    ///   - nClusters: The number of clusters to form as well as the number of centroids to
+    ///     generate, default to 2.
+    ///   - maxIterations: Maximum number of iterations of the k-means algorithm to run, default
+    ///     to 300.
+    ///   - initializer: Select the initialization method for centroids. 'kmean++', 'random'
+    ///     methods for initialization, default to 'kmean++'.
+    ///   - seed: Used to initialize a pseudorandom number generator.
+    public init(
+        nClusters: Int = 2,
+        maxIterations: Int = 300,
+        initializer: String = "kmean++",
+        seed: Int64 = 0
+    ) {
+
+        precondition(nClusters > 1, "Number of clusters must be greater than one.")
+        precondition(maxIterations >= 0, "Maximum number of Iterations must be non-ngative.")
+        precondition(initializer == "kmean++" || initializer == "random",
+            "intialializer must be 'keman++' or 'random'.")
+        
+        self.nClusters = nClusters
+        self.maxIterations = maxIterations
+        self.initializer = initializer
+        self.seed = seed
+        self.centroids = Tensor<Float>([0.0])
+        self.inertia = Tensor<Float>(0)
+        self.labels = Tensor<Int32>([0])
+        
+    }
+    
+    /// Returns the euclidean distance.
+    ///
+    /// - Parameters
+    ///   - a: Tensor<Float> of shape [1, number of features].
+    ///   - b: Tensor<Float> of shape [1, number of features].
+    /// - Returns: The euclidean distace.
+    public func euclideanDistance(a: Tensor<Float>, b: Tensor<Float>) -> Tensor<Float> {
```

Comment:
`euclideanDistance` doesn't depend on any functionality define in `KMeans`.
How about changing it to be a top-level function in `Sources/swiftML/Helpers.swift`?

Please also remove argument labels so that call-sites look more natural:
```
// Currently:
euclideanDistance(a: centroids[i], b: x)
// Goal:
euclideanDistance(centroids[i], x)
```

```suggestion
// Move to Helpers.swift.
public func euclideanDistance(_ a: Tensor<Float>, _ b: Tensor<Float>) -> Tensor<Float> {
```